src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -2828,7 +2828,7 @@ private static DType UnionCore(ref bool fError, DType leftType, DType rightType,
                     ((field1Type == Polymorphic && field2Type.IsRecord) ||
                     (field2Type == Polymorphic && field1Type.IsRecord)))
                 {
-                    fieldType = field1Type == Polymorphic ? field2Type : field1Type;
+                    fieldType = Polymorphic;
```

Comment:
>Polymorphic [](http://example.com/codeflow?start=32&length=11)

Since you're working with polymorphic types, we should add tests for the AsType function. Currently we don't have any tests for that.

Reply:
I do have some on DV repo https://github.com/microsoft/Power-Fx-Dataverse/blob/main/src/PowerFx.Dataverse.Tests/PluginExecutionTests/Functions/AsTypeFunctionTests.cs
I will see if I can add them here...

Reply:
Will add them in seprate PR
https://github.com/microsoft/Power-Fx/issues/2082